### DISCUSSION TITLE
Update frequency of mini sensors

### DISCUSSION THREAD

**Discussion by Rhysgh** [BottlecapDave/HomeAssistant-OctopusEnergy#167] [Q&A] [answered]
Hi @BottlecapDave, first of all, i want to say that I LOVE this integration... 

I see Octopus say the Mini updates every 10 sec; you built in a 1 min refresh for the mini sensors. Can i alter it on my end to try 10-second updates? 

I have had a look, but i couldn't find exactly where you are controlling that.

**Comment by BottlecapDave** [comment, marked as answer]
Hi there and thanks for the positve feedback :)

Based on the latest release (where the location will change in the next release), there are three areas that control the sensors rate
1. The api client controls how data is grouped - https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/v6.0.1/custom_components/octopus_energy/api_client.py#L98
2. The sensors are setup to sync every minute. This applies to **all** sensors. This might have a negative effect on your HA and I'm not sure of the minimum supported frequency - https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/v6.0.1/custom_components/octopus_energy/sensor.py#L49
3. The data is setup to be retrieved and shared every minute - https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/v6.0.1/custom_components/octopus_energy/sensor.py#L112

I set it to this frequency to match the app's refresh rate so that the APIs didn't get flooded with requests, but still be useful.

**Comment by cspiby** [comment]
Would a default of 30s maybe be a little bit better for data resolution? Some consumption could be missed over the course of 60s

**Reply by BottlecapDave** [reply]
Sorry for the late response. No data is lost with the sensors as it retrieves all consumption over the 60s period every 60s. If a 60s update is missed, then it backfills up to 1 hour.

I've set the frequency to match the frequency of updates that the app produces as I don't want to produce any further strain on the API and get it removed. I also didn't want the integration to potentially cause any degrading on a users HA instance from producing too many requests. I have no intention of updating this frequency.
